src/createAction.js

```diff
@@ -10,35 +9,30 @@ export default function createAction(type, payloadCreator = identity, metaCreato
     'Expected payloadCreator to be a function, undefined or null'
   );
 
-  const finalPayloadCreator = isNull(payloadCreator)
+  const finalPayloadCreator = payloadCreator === null || payloadCreator === identity
     ? identity
-    : payloadCreator;
+    : (head, ...args) => head instanceof Error ?
+        head : payloadCreator(head, ...args);
+  
+  const hasMeta = isFunction(metaCreator);
 
   const actionCreator = (...args) => {
-    const hasError = args[0] instanceof Error;
-
-    const action = {
-      type
-    };
-
-    const payload = hasError ? args[0] : finalPayloadCreator(...args);
-    if (!isUndefined(payload)) {
+    const payload = finalPayloadCreator(...args);
+    const action = { type, error: payload instanceof Error };
+    
+    if (payload !== undefined) {
```

Comment:
Wait what's the objection to using `isUndefined`? Does it cost that much bytes that the readability isn't worth it? It's not much less tokens to parse, but still.